### PR TITLE
Add available gas and user args

### PR DESCRIPTION
**Pull request by MaksymMalicki**
This PR introduces the following changes:
1. Passing the input arguments for the main func and initial gas through the flags
2. Parsing of user arguments in form of single numeric values and non-nested numeric arrays, i. e. 1 1 [1 2 3] 4 5 [6] 7 [8 9] + tests
3. Modifications of integration tests pipeline
4. Writing the args to the memory, using ExternalWriteArgsToMemory hint
5. Correction in the main function offset calculation (always 0 for cairo1)
6. Writing gas costs segment to the end of the program segment
7. Correction in calculation of the offset of the `CALL` opcode in the entry code (incremented by the offset of the main fuction)
8. Always writing user args to the `AP`, instead of `AP - len(userArgs)`

### CODE REVIEW

File: pkg/parsers/starknet/starknet.go [hunk 1]
```diff
@@ -95,3 +97,41 @@ func StarknetProgramFromJSON(content json.RawMessage) (*StarknetProgram, error)
 	var starknet StarknetProgram
 	return &starknet, json.Unmarshal(content, &starknet)
 }
+
+type CairoFuncArgs struct {
+	Single *fp.Element
+	Array  []fp.Element
+}
+
+func ParseCairoProgramArgs(input string) ([]CairoFuncArgs, error) {
```
[the remark this answers is not in the document]
Comment (MaksymMalicki) [reply pkg/parsers/starknet/starknet.go:106]: yep, done 🫡

File: pkg/parsers/starknet/starknet.go [hunk 2]
```diff
@@ -95,3 +97,41 @@ func StarknetProgramFromJSON(content json.RawMessage) (*StarknetProgram, error)
...
+				single, err := new(fp.Element).SetString(part)
+				if err != nil {
+					return nil, fmt.Errorf("invalid felt value in array: %v", err)
+				}
+				array = append(array, *single)
+			}
+			result = append(result, CairoFuncArgs{
+				Single: nil,
+				Array:  array,
+			})
```
[the remark this answers is not in the document]
Comment (MaksymMalicki) [reply pkg/parsers/starknet/starknet.go:130]: done 🫡

File: pkg/hintrunner/core/hint.go [hunk 3]
```diff
@@ -1929,3 +1929,38 @@ func (hint *FieldSqrt) Execute(vm *VM.VirtualMachine, _ *hinter.HintRunnerContex
...
+	userArgs, ok := userArgsVar.([]starknet.CairoFuncArgs)
+	if !ok {
+		return fmt.Errorf("expected user args to be a list of CairoFuncArgs")
+	}
+	for _, arg := range userArgs {
+		if arg.Single != nil {
+			mv := mem.MemoryValueFromFieldElement(arg.Single)
+			err := vm.Memory.Write(1, vm.Context.Ap-uint64(len(userArgs)), &mv)
+			if err != nil {
+				return fmt.Errorf("write single arg: %v", err)
```
[the remark this answers is not in the document]
Comment (MaksymMalicki) [reply pkg/hintrunner/core/hint.go:1953]: The error already contains information about the value, as it is returned by the `Write` method

**Review by cicr99 — COMMENTED** [review]
looks great!
The CI is failing though

**Review by cicr99 — APPROVED** [review]

**Pull request merged**